packages/server/src/alerts/alerts.entity.ts

```diff
@@ -0,0 +1,46 @@
+import { AlertPayload, AlertType } from '@koh/common';
+import { Exclude } from 'class-transformer';
+import {
+  BaseEntity,
+  Column,
+  Entity,
+  JoinColumn,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+import { CourseModel } from '../course/course.entity';
+import { UserModel } from '../profile/user.entity';
+
+@Entity('alert_model')
+export class AlertModel extends BaseEntity {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column({ type: 'enum', enum: AlertType })
+  alertType: AlertType;
+
+  @Column()
+  sent: Date;
+
+  @Column({ nullable: true })
+  resolved: Date;
+
+  @ManyToOne((type) => UserModel, (user) => user.alerts)
+  @JoinColumn({ name: 'userId' })
+  user: UserModel;
+
+  @Column({ nullable: true })
+  @Exclude()
+  userId: number;
+
+  @ManyToOne((type) => CourseModel, (course) => course.alerts)
+  @JoinColumn({ name: 'courseId' })
+  course: CourseModel;
+
+  @Column({ nullable: true })
+  @Exclude()
+  courseId: number;
```

Comment:
Why do we need a course relation for an alert?

Reply:
I think so because we want the alerts to be persistent in every part of the app (since it's in the nav bar). Alerts also are meant to be more generic and could apply to non queue specific things ex. cancelled office hours